service/authenticator_test.go

```diff
@@ -1,33 +1,710 @@
 package service
 
 import (
+	"encoding/base64"
+	"github.com/Pallinder/go-randomdata"
+	_ "github.com/golang-migrate/migrate/v4/source/file"
+	pblib "github.com/hwsc-org/hwsc-api-blocks/protobuf/lib"
+	"github.com/hwsc-org/hwsc-app-gateway-svc/consts"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc/metadata"
 	"testing"
 )
 
 func TestAuthenticate(t *testing.T) {
+	/*
+		Test for basic authentication
+	*/
+	baseAuthCase := "test authenticating valid user"
+	validEmail := randomdata.Email()
+	validPassword := "Abcd!123@"
+	resp, errCreateUser := userSvc.createUser(
+		&pblib.User{
+			FirstName:    randomdata.FirstName(randomdata.Male),
+			LastName:     randomdata.LastName(),
+			Email:        validEmail,
+			Password:     validPassword,
+			Organization: testOrg,
+		})
+	assert.Nil(t, errCreateUser, baseAuthCase)
+	assert.NotNil(t, resp, baseAuthCase)
+	emailToken := resp.GetIdentification().GetToken()
+	errVerifyEmailToken := userSvc.verifyEmailToken(emailToken)
+	assert.Nil(t, errVerifyEmailToken, baseAuthCase)
+	validEnc := base64.StdEncoding.EncodeToString([]byte(validEmail + ":" + validPassword))
 
+	header := metadata.New(map[string]string{
+		consts.StrMdBasicAuthHeader: consts.StrBasicAuthPrefix + validEnc,
+	})
+	validIncomingCtx := metadata.NewIncomingContext(context.Background(), header)
+
+	incomingCtx1, errAuthenticate := Authenticate(validIncomingCtx)
+	assert.Nil(t, errAuthenticate, baseAuthCase)
+	incomingMd1, incomingMdOk1 := metadata.FromIncomingContext(incomingCtx1)
+	assert.True(t, incomingMdOk1, baseAuthCase)
+	incomingAuthToken1, incomingAuthTokenOk1 := incomingMd1[consts.StrMdAuthToken]
+	assert.True(t, incomingAuthTokenOk1, baseAuthCase)
+	assert.Equal(t, 1, len(incomingAuthToken1), baseAuthCase)
```

Comment:
Line 19 - 48 sort of shows the registration to login process
1. create user
2. verify email token
3. login

Reply:
Line 48 grabs the auth token string needed by the browser to make GRPC calls

Reply:
You should put these comment inside the actual files, because it is kind of hard to read without them